engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/JobPreparationActor.scala

```diff
@@ -59,15 +60,43 @@ case class JobPreparationActor(executionData: WorkflowExecutionActorData,
   }
 
   def prepareJobExecutionActor(inputs: Map[LocallyQualifiedName, WdlValue]): Try[BackendJobPreparationSucceeded] = {
-    val jobDescriptor = BackendJobDescriptor(workflowDescriptor.backendDescriptor, jobKey, inputs)
-    Try(factory.jobExecutionActorProps(jobDescriptor, initializationData, serviceRegistryActor)) map { BackendJobPreparationSucceeded(jobDescriptor, _) }
+    for {
+      evaluatedRuntimeAtts <- evaluateRuntimeAttributes(jobKey, expressionLanguageFunctions, inputs)
+      jobDescriptor = BackendJobDescriptor(workflowDescriptor.backendDescriptor, jobKey, evaluatedRuntimeAtts, inputs)
+    } yield BackendJobPreparationSucceeded(jobDescriptor, factory.jobExecutionActorProps(jobDescriptor, initializationData, serviceRegistryActor))
   }
 
   // Split inputs map (= evaluated workflow declarations + coerced json inputs) into [init\.*].last
   private lazy val splitInputs = workflowDescriptor.backendDescriptor.inputs map { case (fqn, v) => splitFqn(fqn) -> v }
 
-  def resolveAndEvaluate(jobKey: BackendJobDescriptorKey,
-                         wdlFunctions: WdlStandardLibraryFunctions): Try[Map[LocallyQualifiedName, WdlValue]] = {
+  def evaluateRuntimeAttributes(jobKey: BackendJobDescriptorKey,
+                                wdlFunctions: WdlStandardLibraryFunctions,
+                                evaluatedInputs: Map[LocallyQualifiedName, WdlValue]): Try[Map[String, WdlValue]] = {
+    val tryInputs = evaluatedInputs.mapValues(Try(_))
```

Comment:
I think it's fine here (but please double check), but keep in mind that mapValues doesn't actually transform the variable but creates a view. So every time it is referenced that's reapplied